### ISSUE TITLE
Invisible cursor and strange colors

### ISSUE THREAD

**Issue by damanis** [hismailbulut/Neoray#35]
**OS**
  Ubuntu 20.04

**Neoray version**
  Neoray version: compiled [51f01a12](https://github.com/hismailbulut/Neoray/commit/51f01a12d66dd05191992c17b13188f292a008bb)
  nvim version:
    NVIM v0.8.0-dev+608-g778541067

**Problems**
- Sometimes a cursor still invisible
- Sometimes strange colors:
![image](https://user-images.githubusercontent.com/65733488/178741170-ef842f59-16cc-4afa-9f52-f1c361f1533c.png)
instead of
![image](https://user-images.githubusercontent.com/65733488/178741573-e50a4cca-1037-4384-b549-5c4055ca545b.png)
If return to version 0.1.1 - same problem. Before trying this version, 0.1.1 worked properly.

**Steps to reproduce**
1. Create file:
```
TITLE: qweqwe
END: qweqwe

02-01-2022: text2

01-01-2022: text1
```
2. Add to some nvim lua init file:
```
vim.api.nvim_exec([[
au BufNewFile,BufRead *.txt call Detect_jdoc()
fun Detect_jdoc()
    if getline(1) =~? "^TITLE:" || getline(2) =~? "^TITLE:"
        setlocal filetype=jdoc
    endif
endfun
]], false)
```
3. Open the created file by Neoray several times.

**Neoray config in init.vim**
lua/settings.lua:
```
vim.api.nvim_exec([[
if exists('g:neoray')
    " Now impossible to disable cursor animation by 'NeoraySet CursorAnimTime 0'
    NeoraySet CursorAnimTime 0.001
    NeoraySet ContextMenuOn false
    NeoraySet WindowSize 100x52
    colorscheme uwu
    set guifont=Go-Mono-Nerd-Font-Complete:h11
endif
]], false)
```

**Debug output (you can generate starting neoray with --verbose option)**
The is no difference between Neoray_verbose.log in success and fail

**Comment by hismailbulut** [comment]
Thanks for detailed information.
I tried but I couldn't reproduce this. Is this started happening after 51f01a1 ?
Also this can be a colorscheme issue. If you updated your plugins, a bug in a plugin may also have caused this.

**Comment by damanis** [comment]
I didn't see the problem until try [51f01a12](https://github.com/hismailbulut/Neoray/commit/51f01a12d66dd05191992c17b13188f292a008bb).
Checked with different colorschemes (for example, [github-nvim-theme](https://github.com/projekt0n/github-nvim-theme)). Others neovim GUI I use (fvim, neovim-qt, goneovim, nvui) have do problem with same configuration.

**Comment by hismailbulut** [comment]
So it seems this is not a Neoray issue. This is may be issue of neovim itself or some plugins you are using (I don't know if you are using related plugins for jdoc but it also may cause this) This may remain open until you find out exactly where the issue is coming from.

**Comment by damanis** [comment]
I checked now with neovim 0.7.2 - same behavior. It doesn't seems nvim issue.
I'll update if I find something new.

**Comment by damanis** [comment]
@hismailbulut 
It not about the issue.
Now ```NeoraySet CursorAnimTime 0``` does not work. Is there a way to disable cursor animation?

**Comment by damanis** [comment]
Does Neoray saves some internal files (states, configs)?

**Comment by hismailbulut** [comment]
> Does Neoray saves some internal files (states, configs)?

No, Neoray only saves some debugging files in debug build (cpu and heap profile, crash log and verbose log)
Neoray never saves any config or never changes your any file in your system

**Comment by damanis** [comment]
@hismailbulut 
Do you use (during development) some debug options of GLFW? I still can't catch the problem with cursor.

**Comment by hismailbulut** [comment]
There are some ways you can debug this behavior
Please do every step

### Debug from Neoray
1. First build neoray with debug tag, just run `make build` in the workspace and it will generate neoray_debug in bin folder
2. Open debug build, reproduce issue and middle click one of the cell has issue. This will print cell info to terminal
3. Copy cell info and put it here so I can look at it

### Debug from neovim
1. Move cursor to the same cell you middle clicked
2. Run command `:echo synIDattr(synIDtrans(synID(line("."), col("."), 1)), "fg")` and put output here
3. Run command `:echo synIDattr(synIDtrans(synID(line("."), col("."), 1)), "bg")` and put output here
4. Run command `:echo synIDattr(synIDtrans(synID(line("."), col("."), 1)), "name")` and put output here
5. Run command `:echo synIDattr(synID(line("."), col("."), 1), "name")` and put output here

### Additional steps
1. Open Neoray with command `./neoray_debug -u NORC` and try to reproduce
2. Open Neoray with command './neoray_debug --noplugin' and try to reproduce

**Comment by damanis** [comment]
**Debug from Neoray**
[TRACE] GLFW3 Version: 3.3.4 X11 GLX EGL OSMesa clock_gettime evdev
[TRACE] Opengl Version: 4.5 (Core Profile) Mesa 21.2.6
[TRACE] Vendor: Mesa/X.org
[TRACE] Renderer: llvmpipe (LLVM 12.0.0, 256 bits)
[TRACE] GLSL: 4.50
[TRACE] Max Texture Size: 16384
[TRACE] Neovim started with command: nvim [--embed /home/sergeyp/.bashrc]
[TRACE] Neovim version 0.8.0
[DEBUG] Calculated startup size of the neovim is 31 88
[DEBUG] Attached to neovim as an ui client
[TRACE] Initialization time: 201.510109ms
[DEBUG] Grid created: Grid(ID: 1, Number: 1, Type: Normal, Pos: 0 0, PixelPos: common.Vector2[int](X: 0, Y: 0), Size: 31 88, Window: 0, Hidden: false)
[TRACE] Loading font Go-Mono-Nerd-Font-Complete
[TRACE] Regular: /home/user/.fonts/Go-Mono-Nerd-Font-Complete.ttf
[TRACE] Bold: /home/user/.fonts/Go-Mono-Nerd-Font-Complete-Bold.ttf
[TRACE] Italic: /home/user/.fonts/Go-Mono-Nerd-Font-Complete-Italic.ttf
[TRACE] BoldItalic: /home/user/.fonts/Go-Mono-Nerd-Font-Complete-Bold-Italic.ttf
[DEBUG] Option CursorAnimTime is 0
[DEBUG] Option ContextMenuOn is false
[DEBUG] Option WindowSize is 100 52
[TRACE] Window is visible now in 805.488448ms
[DEBUG] Mouse action: press 1 0 0
[DEBUG] CellInfo (1, 0, 0): Grid(ID: 1, Number: 1, Type: Normal, Pos: 0 0, PixelPos: common.Vector2[int](X: 0, Y: 0), Size: 52 100, Window: 0, Hidden: false), Cell(Char: 35 #, AttribID: 182), Vertex(pos: common.Rectangle[float32](X: 0, Y: 0, W: 9, H: 15), tex1: common.Rectangle[float32](X: 0, Y: 0, W: 0.0021972656, H: 0.029296875), tex2: common.Rectangle[float32](X: 0, Y: 0, W: 0, H: 0), fg: common.Color[float32](R: 0.42352942, G: 0.42352942, B: 0.42352942, A: 1), bg: common.Color[float32](R: 0.07450981, G: 0.101960786, B: 0.10980392, A: 1), sp: common.Color[float32](R: 0, G: 0, B: 0, A: 0))
[DEBUG] Mouse action: release 1 0 0

**Comment by damanis** [comment]
Commands from section 'Debug from neovim' did not print anything.

**Comment by hismailbulut** [comment]
Does the issue happens when you run with --noplugin?
And those commands has to echo the attributes in neovim command line. Try `:messages` after commands

**Comment by damanis** [comment]
The issue happens both for '--noplugin' and '-u NORC'

**Comment by hismailbulut** [comment]
> Others neovim GUI I use (fvim, neovim-qt, goneovim, nvui) have do problem with same configuration.

Did you mean "no" here

Can you zip and upload a copy of your entire vim config and the file you are working on excluding your sensitive information?

**Comment by damanis** [comment]
> Did you mean "no" here

Sure

**Comment by hismailbulut** [comment]
> Sure

So I'm sorry because I misunderstood you, this is Neoray issue

**Comment by hismailbulut** [comment]
> Debug output didn't help?

They only helpful together with outputs of commands in `Debug from neovim`

**Comment by hismailbulut** [comment]
I can't reproduce with minimal configuration
```
au BufNewFile,BufRead *.txt call Detect_jdoc()

fun Detect_jdoc()
    echomsg "Detecting jdoc"
    if getline(1) =~? "^TITLE:" || getline(2) =~? "^TITLE:"
        echomsg "jdoc detected"
        setlocal filetype=jdoc
    endif
endfun

if exists('g:neoray')
    " Now impossible to disable cursor animation by 'NeoraySet CursorAnimTime 0'
    " NeoraySet CursorAnimTime 0.001
    NeoraySet ContextMenuOn false
    NeoraySet WindowSize 100x52
    " colorscheme uwu
    " set guifont=Go-Mono-Nerd-Font-Complete:h11
endif
```

with start command: `neoray -u issue35.vim --noplugin`

result:
![image](https://user-images.githubusercontent.com/73897211/182177656-1180e84a-2851-4411-a1b8-641d7cd88617.png)

**Comment by damanis** [comment]
I found why 'echo' commands you requested are not shown - their output is empty.

**Comment by damanis** [comment]
@hismailbulut 
Today I compiled source with last your changes (last commit is 0f9133ef) and currently I don't see the invisible cursor problem.
I will update you if it returns.
